In [21]:
import pandas as pd
import numpy as np
import seaborn as sns
import geopandas as gpd
import matplotlib.pyplot as plt

In [22]:
import helper_400

%load_ext autoreload
%autoreload 2
helper_400.set_sns_style()

The autoreload extension is already loaded. To reload it, use:
  %reload_ext autoreload


<Figure size 800x600 with 0 Axes>

In [23]:
#df = pd.read_parquet("outputs/agg_hotspots_new_heat.parquet")

In [24]:
df = gpd.read_file("outputs/d2-events-2d-230929_SVI_shapefile.geojson")

In [25]:
df["TRACT"] = df.GEOID.str.slice(5, 12)

In [26]:
from tabulate import tabulate

In [27]:
df.columns

Index(['ST', 'STATE', 'ST_ABBR', 'STCNTY', 'COUNTY', 'FIPS', 'LOCATION',
       'AREA_SQMI', 'E_TOTPOP', 'M_TOTPOP',
       ...
       'ws5', 'heatday_2d', 'smoke_pm_non_zero_2d', 'wfday_2d', '_hws_2d',
       'hs_2d', 'hws_2d', 'ws_2d', 'geometry', 'TRACT'],
      dtype='object', length=182)

# Average

In [28]:
df = df[df != "-999"]

In [29]:
df[["ws", "hws", "hs", "hw"]].mean().round(1)

ws      2.4
hws     0.1
hs     31.7
hw      0.2
dtype: float64

In [30]:
df[["EP_AGE65","EP_DISABL","EP_MINRTY","EP_POV150","EP_UNINSUR"]].mean().round(1)

EP_AGE65      11.4
EP_DISABL      5.0
EP_MINRTY     48.0
EP_POV150     20.9
EP_UNINSUR     0.8
dtype: float64

# Top CTs

In [31]:
for x in ["ws", "hws", "hs", "hw"]:
    a= df.sort_values(by=x, ascending=False)[["TRACT","COUNTY","ST_ABBR","EP_AGE65","EP_DISABL","EP_MINRTY","EP_POV150","EP_UNINSUR",x]].head(10)
    print(tabulate(a, headers='keys', tablefmt='pipe', showindex=False))

|   TRACT | COUNTY     | ST_ABBR   |   EP_AGE65 |   EP_DISABL |   EP_MINRTY |   EP_POV150 |   EP_UNINSUR |   ws |
|--------:|:-----------|:----------|-----------:|------------:|------------:|------------:|-------------:|-----:|
|  960100 | Idaho      | ID        |       34.5 |        28.6 |         6.3 |        22.7 |         17.7 |  627 |
|  960200 | Caribou    | ID        |       15.7 |        15.5 |         5.7 |        16   |         12.9 |  478 |
|  001300 | Siskiyou   | CA        |       26.3 |        21.1 |        38   |        35.7 |          5.8 |  438 |
|  000800 | Siskiyou   | CA        |       28.9 |        18.7 |        21.7 |        27.7 |          3.6 |  403 |
|  002701 | Tulare     | CA        |       32.4 |        25.7 |        26.6 |        45.9 |          5.2 |  387 |
|  001500 | Coconino   | AZ        |       36.3 |        14.7 |        17.3 |        15.8 |         11.1 |  378 |
|  970100 | Valley     | ID        |       47.7 |        29.4 |         6.7 |        24.

|   TRACT | COUNTY     | ST_ABBR   |   EP_AGE65 |   EP_DISABL |   EP_MINRTY |   EP_POV150 |   EP_UNINSUR |   ws |
|--------:|:-----------|:----------|-----------:|------------:|------------:|------------:|-------------:|-----:|
|  960100 | Idaho      | ID        |       34.5 |        28.6 |         6.3 |        22.7 |         17.7 |  627 |
|  960200 | Caribou    | ID        |       15.7 |        15.5 |         5.7 |        16   |         12.9 |  478 |
|  001300 | Siskiyou   | CA        |       26.3 |        21.1 |        38   |        35.7 |          5.8 |  438 |
|  000800 | Siskiyou   | CA        |       28.9 |        18.7 |        21.7 |        27.7 |          3.6 |  403 |
|  002701 | Tulare     | CA        |       32.4 |        25.7 |        26.6 |        45.9 |          5.2 |  387 |
|  001500 | Coconino   | AZ        |       36.3 |        14.7 |        17.3 |        15.8 |         11.1 |  378 |
|  970100 | Valley     | ID        |       47.7 |        29.4 |         6.7 |        24.8 |          6.5 |  340 |
|  970100 | Clearwater | ID        |       36.6 |        24.4 |         3.8 |        21.4 |         15.7 |  318 |
|  960200 | Custer     | ID        |       33.2 |        27   |        14.8 |        27.2 |          6.2 |  311 |
|  960200 | Shoshone   | ID        |       28.8 |        29.7 |         7   |        29.3 |         13   |  308 |
|   TRACT | COUNTY   | ST_ABBR   |   EP_AGE65 |   EP_DISABL |   EP_MINRTY |   EP_POV150 |   EP_UNINSUR |   hws |
|--------:|:---------|:----------|-----------:|------------:|------------:|------------:|-------------:|------:|
|  964100 | Grant    | NM        |       36.2 |        24.4 |        32.4 |        30.2 |          5.8 |    25 |
|  951700 | Elko     | NV        |       13.7 |        15.2 |        74.4 |        31.4 |         12.5 |    24 |
|  001300 | Siskiyou | CA        |       26.3 |        21.1 |        38   |        35.7 |          5.8 |    22 |
|  210000 | Douglas  | OR        |       29.7 |        22.6 |         9.1 |        29.5 |          9.4 |    22 |
|  970900 | Malheur  | OR        |       13.1 |        13.9 |        29.5 |        21   |         13.6 |    21 |
|  976401 | Catron   | NM        |       37.3 |        24.3 |        26.7 |        53.9 |          3.1 |    21 |
|  960100 | Greenlee | AZ        |       25.4 |        19.6 |        60.4 |        26.3 |          2.2 |    21 |
|  962402 | Sierra   | NM        |       39.4 |        19.2 |        42.5 |        34.3 |          3.4 |    20 |
|  940400 | Gila     | AZ        |        7.9 |        12.7 |        98.2 |        58.6 |          8.3 |    19 |
|  961600 | Graham   | AZ        |        8.2 |        12.7 |        53.7 |        18   |         10.3 |    19 |
|   TRACT | COUNTY     | ST_ABBR   |   EP_AGE65 |   EP_DISABL |   EP_MINRTY |   EP_POV150 |   EP_UNINSUR |   hs |
|--------:|:-----------|:----------|-----------:|------------:|------------:|------------:|-------------:|-----:|
|  960100 | Greenlee   | AZ        |       25.4 |        19.6 |        60.4 |        26.3 |          2.2 |  100 |
|  000400 | Morgan     | CO        |       19.9 |        11.5 |        49.3 |        18.6 |          7.4 |   97 |
|  000500 | Morgan     | CO        |       12.8 |        10.7 |        69.5 |        34.4 |          8.6 |   97 |
|  000300 | Morgan     | CO        |       16.6 |        11.3 |        42.6 |        18.2 |         12.8 |   95 |
|  000600 | Morgan     | CO        |       10.1 |         7.4 |        59.8 |        25.3 |         10.7 |   92 |
|  000800 | Las Animas | CO        |       20   |        19.3 |        40.1 |        23   |          5.6 |   92 |
|  962402 | Sierra     | NM        |       39.4 |        19.2 |        42.5 |        34.3 |          3.4 |   90 |
|  964200 | Grant      | NM        |       35.5 |        21.7 |        25.4 |        26.9 |          6.5 |   89 |
|  962100 | Kit Carson | CO        |       17.2 |        14   |        27.2 |        22.7 |         12   |   87 |
|  000700 | Morgan     | CO        |       14.1 |        11.8 |        26.4 |        22.1 |          8.9 |   86 |
|   TRACT | COUNTY   | ST_ABBR   |   EP_AGE65 |   EP_DISABL |   EP_MINRTY |   EP_POV150 |   EP_UNINSUR |   hw |
|--------:|:---------|:----------|-----------:|------------:|------------:|------------:|-------------:|-----:|
|  000800 | Gila     | AZ        |       31   |        29.4 |        46.5 |        51.7 |          3.5 |   44 |
|  001300 | Gila     | AZ        |       20.5 |        22.8 |        73.7 |        43.7 |          7.6 |   38 |
|  964100 | Grant    | NM        |       36.2 |        24.4 |        32.4 |        30.2 |          5.8 |   27 |
|  970900 | Malheur  | OR        |       13.1 |        13.9 |        29.5 |        21   |         13.6 |   27 |
|  210000 | Douglas  | OR        |       29.7 |        22.6 |         9.1 |        29.5 |          9.4 |   26 |
|  960100 | Greenlee | AZ        |       25.4 |        19.6 |        60.4 |        26.3 |          2.2 |   25 |
|  001500 | Coconino | AZ        |       36.3 |        14.7 |        17.3 |        15.8 |         11.1 |   25 |
|  951700 | Elko     | NV        |       13.7 |        15.2 |        74.4 |        31.4 |         12.5 |   25 |
|  961600 | Graham   | AZ        |        8.2 |        12.7 |        53.7 |        18   |         10.3 |   24 |
|  976401 | Catron   | NM        |       37.3 |        24.3 |        26.7 |        53.9 |          3.1 |   23 |
|   TRACT | COUNTY     | ST_ABBR   |   EP_AGE65 |   EP_DISABL |   EP_MINRTY |   EP_POV150 |   EP_UNINSUR |   ws |
|--------:|:-----------|:----------|-----------:|------------:|------------:|------------:|-------------:|-----:|
|  960100 | Idaho      | ID        |       34.5 |        28.6 |         6.3 |        22.7 |         17.7 |  627 |
|  960200 | Caribou    | ID        |       15.7 |        15.5 |         5.7 |        16   |         12.9 |  478 |
|  001300 | Siskiyou   | CA        |       26.3 |        21.1 |        38   |        35.7 |          5.8 |  438 |
|  000800 | Siskiyou   | CA        |       28.9 |        18.7 |        21.7 |        27.7 |          3.6 |  403 |
|  002701 | Tulare     | CA        |       32.4 |        25.7 |        26.6 |        45.9 |          5.2 |  387 |
|  001500 | Coconino   | AZ        |       36.3 |        14.7 |        17.3 |        15.8 |         11.1 |  378 |
|  970100 | Valley     | ID        |       47.7 |        29.4 |         6.7 |        24.8 |          6.5 |  340 |
|  970100 | Clearwater | ID        |       36.6 |        24.4 |         3.8 |        21.4 |         15.7 |  318 |
|  960200 | Custer     | ID        |       33.2 |        27   |        14.8 |        27.2 |          6.2 |  311 |
|  960200 | Shoshone   | ID        |       28.8 |        29.7 |         7   |        29.3 |         13   |  308 |